src/Protocol/Diaspora.php

```diff
@@ -3939,104 +3942,100 @@ public static function splitName(string $name): array
 				[$year, $month, $day] = sscanf($profile['dob'], '%4d-%2d-%2d');
 				if ($year < 1004) {
 					$year = 1004;
 				}
-				$dob = DateTimeFormat::utc($year . '-' . $month . '-'. $day, 'Y-m-d');
+				$data['birthday'] = DateTimeFormat::utc($year . '-' . $month . '-' . $day, 'Y-m-d');
 			}
 
-			$about = BBCode::toMarkdown($profile['about']);
+			$data['about'] = BBCode::toMarkdown($profile['about']);
+
+			$data['location'] = $profile['location'];
+			$data['tag_string'] = '';
 
-			$location = $profile['location'];
-			$tags = '';
 			if ($profile['pub_keywords']) {
 				$kw = str_replace(',', ' ', $profile['pub_keywords']);
 				$kw = str_replace('  ', ' ', $kw);
 				$arr = explode(' ', $kw);
 				if (count($arr)) {
 					for ($x = 0; $x < 5; $x ++) {
 						if (!empty($arr[$x])) {
-							$tags .= '#'. trim($arr[$x]) .' ';
+							$data['tag_string'] .= '#'. trim($arr[$x]) .' ';
 						}
 					}
 				}
 			}
-			$tags = trim($tags);
+			$data['tag_string'] = trim($data['tag_string']);
 		}
 
-		return [
-			'author' => $handle,
-			'first_name' => $first,
-			'last_name' => $last,
-			'image_url' => $large,
-			'image_url_medium' => $medium,
-			'image_url_small' => $small,
-			'birthday' => $dob,
-			'bio' => $about,
-			'location' => $location,
-			'searchable' => $searchable,
-			'nsfw' => 'false',
-			'tag_string' => $tags
-		];
+		return $data;
 	}
 
 	/**
 	 * Sends profile data
 	 *
-	 * @param int   $uid    The user id
-	 * @param array $recips optional, default empty array
+	 * @param int   $uid        The user id
+	 * @param array $recipients optional, default empty array
+	 *
 	 * @return void
 	 * @throws \Exception
 	 */
-	public static function sendProfile(int $uid, array $recips = [])
+	public static function sendProfile(int $uid, array $recipients = [])
 	{
 		if (!$uid) {
+			Logger::warning('Parameter "uid" is empty');
 			return;
 		}
 
 		$owner = User::getOwnerDataById($uid);
-		if (!$owner) {
+		if (empty($owner)) {
+			Logger::warning('Cannot fetch User record', ['uid' => $uid]);
 			return;
 		}
 
-		if (!$recips) {
-			$recips = DBA::selectToArray('contact', [], ['network' => Protocol::DIASPORA, 'uid' => $uid, 'rel' => [Contact::FOLLOWER, Contact::FRIEND]]);
+		if (!$recipients) {
+			Logger::debug('No recipients provided, fetching for user', ['uid' => $uid]);
+			$recipients = DBA::selectToArray('contact', [], ['network' => Protocol::DIASPORA, 'uid' => $uid, 'rel' => [Contact::FOLLOWER, Contact::FRIEND]]);
 		}
 
-		if (!$recips) {
+		if (!$recipients) {
+			Logger::warning('Cannot fetch recipients', ['uid' => $uid]);
 			return;
 		}
 
 		$message = self::createProfileData($uid);
 
-		// @ToDo Split this into single worker jobs
-		foreach ($recips as $recip) {
-			Logger::info("Send updated profile data for user " . $uid . " to contact " . $recip['id']);
-			self::buildAndTransmit($owner, $recip, 'profile', $message);
+		// @todo Split this into single worker jobs
+		foreach ($recipientients as $recipient) {
```

Comment:
This variable doesn't exist in the current function. I recommend using an IDE that performs syntax checks to avoid this.

Reply:
I usually write PHP code with an editor.

Reply:
I can suggest VS Code with an PHP extension.

Reply:
We need more unit tests that tests our code, then you can edit it with an editor that has syntax-highlighting.

Reply:
Unit tests alone cannot fix this. But it would be fine if there was some routine that performs the same code checks like the VS Code PHP extension or phpStorm that checks every line of code.

Reply:
These tools can only make limited assumptions about the data types since it can't guess data types from the database or the hooks for examples.